src/pass/storage_flatten.cc

```diff
@@ -520,9 +529,20 @@ class StorageFlattener : public IRMutator {
 
 Stmt StorageFlatten(Stmt stmt, Map<Tensor, Buffer> extern_buffer,
                     int cache_line_size, bool create_bound_attributes) {
+  /*
+   * Unforunately we have to resort to shared_ptr because Analyzer used by
+   * bounded_analyzer has other analyzers in it. e.g. canonical.
+   * These ones allocate impl and destroy them on destructor calls.
+   * However there is no copy/move operator on analyzer that safely copies
+   * or moves data. Perhaps we should disable copy operator and implement
+   * move operator.
+   */
```

Comment:
@tqchen, btw in case you missed this comment of mine here, I think we would want to file task/issue to fix Analyzer so as to not have to resort to shared_ptr.

Reply:
```
IRVisitorWithAnalyzer analyzer

StorageFlattener(..., &analyzer,).Mutate()

```

Can we simply pass in raw pointer of IRVisitorWithAnalyzer* into StorageFlattener? given that we do not need to share the life-cycles between here and the flattener. 

disable copy constructor sounds good